BrowserKit/Tests/ReduxTests/Utilities/FakeReduxState.swift

```diff
@@ -11,15 +11,17 @@ struct FakeReduxState: StateType, Equatable {
     var isInPrivateMode = false
 
     static let reducer: Reducer<Self> = { state, action in
-        switch action {
-        case FakeReduxAction.initialValueLoaded(let value),
-            FakeReduxAction.counterIncreased(let value),
-            FakeReduxAction.counterDecreased(let value):
-            return FakeReduxState(counter: value,
+        guard let action = action as? FakeReduxAction else { return state }
```

Comment:
One minor concern: it seems like with this change, reducers/middlewares that are interested in handling several different Action subclasses are going to have to check the class type of the incoming action against every subclass they want to handle 1 or more types for. Not a big deal, but that might lead to some clunky boilerplate of its own since we'll need to have multiple switches against the `.actionType` for each individual Action subclass, wrapped by if statements. (Hopefully that made sense the way I phrased it)

Reply:
I agree this is a concern. It touches on one of the other actions items I have to follow up on and that is how we have broken down and organized our states/middlewares and reducers. They are too big and broad right now, we have some switch statements that are already hundreds of lines long. 
Ideally a reducers that wants to reduce actions from more than one Action subclass should be the exception rather than the norm. I'm hoping that will avoid this approach turning into a mess of casting attempts and fallbacks. Open to discussing further if you think that won't be enough.